ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/vulnerabilities/index.tsx

```diff
@@ -28,16 +45,155 @@ import {
+import { getVulnerabilityRatingBackgroundColor } from '@/helpers/get-status-colors';
+import { paginationSchema } from '@/schemas';
+
+const defaultPageSize = 10;
+
+const columns: ColumnDef<VulnerabilityWithIdentifier>[] = [
+  {
+    accessorKey: 'vulnerability',
+    header: () => <></>,
+    cell: ({ row }) => {
+      const type = row.original.identifier.type;
+      const namespace = row.original.identifier.namespace;
+      const name = row.original.identifier.name;
+      const version = row.original.identifier.version;
+      const references = row.original.vulnerability.references;
+
+      // Calculate the overall vulnerability rating based on the individual ratings
+      const ratings = row.original.vulnerability.references.map((reference) =>
+        Number(reference.severity)
+      );
+      const overallRating = calcOverallVulnerability(ratings);
+
+      return (
+        <div className='flex flex-col gap-2'>
+          <div className='flex justify-between'>
+            <Label className='font-semibold'>
+              {type ? type.concat(':') : ''}
+              {namespace ? namespace.concat('/') : ''}
+              {name ? name : ''}
+              {version ? '@'.concat(version) : ''}
+            </Label>
+
+            <Badge className='bg-gray-400'>
+              {row.original.vulnerability.externalId}
+            </Badge>
+          </div>
+          <div className='flex justify-between'>
+            <div>{row.original.vulnerability.summary}</div>
+            <Badge
+              className={`${getVulnerabilityRatingBackgroundColor(overallRating)}`}
+            >
+              {overallRating}
+            </Badge>
+          </div>
+          <div className='italic text-muted-foreground'>
+            {row.original.vulnerability.description}
+          </div>
+          <Accordion type='single' collapsible className='ml-4'>
+            <AccordionItem
+              value='references'
+              className='rounded-lg border px-2'
+            >
+              <AccordionTrigger>
+                <Label className='font-semibold text-blue-400 hover:underline'>
+                  References
+                </Label>
+              </AccordionTrigger>
+              {references.map((reference, index) => (
+                <AccordionContent key={index}>
+                  <div className='flex gap-2'>
+                    <div>{reference.severity}</div>
+                    <div>{reference.scoringSystem}</div>
+                    <Link
+                      className='font-semibold text-blue-400 hover:underline'
+                      to={reference.url}
+                      target='_blank'
+                    >
+                      {reference.url}
+                    </Link>
+                  </div>
+                </AccordionContent>
+              ))}
+            </AccordionItem>
+          </Accordion>
+        </div>
+      );
+    },
+  },
+];
```

Comment:
I thought you asked about table as a concept, not about TanStack table specifically. 🤓

Reply:
This could be changed into better table format in a follow-up PR, where at least the Id, Severity rating and Package could be sortable columns, maybe some of them filterable, too.

Reply:
I think using `table` like this creates some accessibility issues at least, as it likely renders an empty `<thead><tr><th></th></tr></thead>` and the `<td>`s of the single column have so much information. If we want to build this with free-form divs, it could make sense to abstract [`DataTablePagination`](https://github.com/eclipse-apoapsis/ort-server/blob/main/ui/src/components/data-table/data-table-pagination.tsx) to take in the parameters individually without requiring `table` and use it for this kind of lists as well.